Admin auth content review11 #8361 ke (#616)
SUSE/doc-susemanager: modules/administration/pages/auth-methods.adoc

= Authentication Methods

== Authentication Via PAM

> As security measures become increasingly complex, {productname} supports network-based authentication systems via Pluggable Authentication Modules (PAM).

Reviewer: I think the first part of this sentence is needless, it's kinda just making a general assumption about the world. It's more authoritative to just launch straight in:

```suggestion
{productname} supports network-based authentication systems via Pluggable Authentication Modules (PAM).
```

> PAM is a suite of libraries that allows to integrate {productname} with a centralized authentication mechanism, thus eliminating the need to remember multiple passwords.

Reviewer: ```suggestion
PAM is a suite of libraries that allows you to integrate {productname} with a centralized authentication mechanism, eliminating the need to remember multiple passwords.
```

Reviewer: Do people use PAM so that they don't need to remember passwords? Isn't it more about connecting to a workplace user db?

{productname} supports LDAP, Kerberos, and other network-based authentication systems via PAM.

> === Enabling PAM

Reviewer: ```suggestion
=== Enable PAM
```

> To enable {productname} to use PAM in your organization's authentication infrastructure, follow the steps below.

Reviewer: This sentence is redundant.

Author: ok, submission pending...

> . Set up a PAM service file (default location: [path]``/etc/pam.d/susemanager``).

Reviewer: ```suggestion
. Create a PAM service file.
PAM service files are located in [path]``/etc/pam.d/susemanager`` by default.
```

> Enforce its use by adding the following line to [path]``/etc/rhn/rhn.conf``:

Reviewer: ```suggestion
Enforce its use by adding this line to [path]``/etc/rhn/rhn.conf``:
```

+

----
pam_auth_service = susemanager
----
+
NOTE: This assumes the PAM service file is named [path]``susemanager``.
+

> . Navigate to the menu:Create User[] page to enable a new or existing user to authenticate with PAM

Reviewer: ```suggestion
. In the {productname} {webui}, navigate to menu:Create User[] and enable a new or existing user to authenticate with PAM
```

This could probably be clearer.

> . Select the checkbox labeled [guimenu]``Pluggable Authentication Modules (PAM)`` positioned below the password and password confirmation fields.

Reviewer: ```suggestion
. Check the [guimenu]``Pluggable Authentication Modules (PAM)`` checkbox.
It is below the password and password confirmation fields.
```

> . To authenticate a {slsa} system against Kerberos add the following lines to [path]``/etc/pam.d/susemanager``:

Reviewer: ```suggestion
. To authenticate a {slsa} system against Kerberos, add these lines to [path]``/etc/pam.d/susemanager``:
```

+

----
#%PAM-1.0
auth     include        common-auth
account  include        common-account
password include        common-password
session  include        common-session
----
+

> NOTE: To register a Red Hat Enterprise Linux System with Kerberos add the following lines to [path]``/etc/pam.d/susemanager``:

Reviewer: ```suggestion
To authenticate a {rhel} system against Kerberos, add these lines to [path]``/etc/pam.d/susemanager``:
```

----
#%PAM-1.0
auth        required      pam_env.so
auth        sufficient    pam_krb5.so no_user_check
auth        required      pam_deny.so
account     required      pam_krb5.so no_user_check
----
+

> YaST can now be used to configure PAM, when packages such as [package]``yast2-ldap-client`` and [package]``yast2-kerberos-client`` are installed.

Reviewer: ```suggestion
YaST can be used to configure PAM.
You will need to install the [package]``yast2-ldap-client`` and [package]``yast2-kerberos-client`` packages.
```

> For information about configuring PAM, see the SUSE Linux Enterprise Server Security Guide https://www.suse.com/documentation/sles-15/book_security/data/part_auth.html.

Reviewer: ```suggestion
For more information about configuring PAM, see the SUSE Linux Enterprise Server Security Guide https://www.suse.com/documentation/sles-15/book_security/data/part_auth.html.
```

> This example is not limited to Kerberos; it is generic and uses the current server configuration.

Reviewer: ```suggestion
This is a generic example that will also work for other network-based authentication methods.
```

> Only network based authentication services are supported.

Reviewer: ```suggestion

```

+
.Changing Passwords

> IMPORTANT: Changing the password on the {productname} {webui} changes only the local password on the {productname} server.

Reviewer: ```suggestion
IMPORTANT: Changing the password on the {productname} {webui} changes only the local password on the {productname} Server.
```

> This password may not be used at all if PAM is enabled for that user.

Reviewer: ```suggestion
If PAM is enabled for that user, the password might not be used at all.
```

In the above example, for instance, the Kerberos password will not be changed.

> === Authentication Via eDirectory and PAM

Reviewer: ```suggestion
=== Authentication with eDirectory and PAM
```

. First check to ensure eDirectory authentication is working with your current OS for example:
+

----
getent passwd
----
. If users are returned from eDirectory then create the [filename]``/etc/pam.d/susemanager`` and add the following content:
+

----
#%PAM-1.0
auth     include        common-auth
account  include        common-account
password include        common-password
session  include        common-session
----

> . Finally add the following line to the {productname} conf file [filename]``/etc/rhn/rhn.conf``:

Reviewer: ```suggestion
. Add these lines to the {productname} configuration file at [filename]``/etc/rhn/rhn.conf``:
```

+

----
pam_auth_service = susemanager
----

> You may now create users with the same id that appears on eDirectory and enable the [guimenu]``Use PAM`` check-box in the {productname} {webui}.

Reviewer: ```suggestion
You can now create users with the same ID used by eDirectory, and check the [guimenu]``Use PAM`` checkbox in the {productname} {webui}.
```

> === Example Quest VAS Active Directory Authentication Template

Reviewer: You could probably do this as `.Example:` rather than having a third-level heading.

Author: sounds good. done.

> If you are using Quest VAS for active directory authentication, you can use the following [path]``/etc/pam.d/susemanager`` file.

Reviewer: ```suggestion
If you are using Quest VAS for active directory authentication, you can use this [path]``/etc/pam.d/susemanager`` file.
```

* Log out with service provider-initiated single logout (SLO), and Identity service provider single logout service (SLS).
* Assertion and nameId encryption.
* Assertion signatures.
* Message signatures with AuthNRequest, LogoutRequest, and LogoutResponses.
* Enable an Assertion consumer service endpoint.
* Enable a single logout service endpoint.
* Publish the SP metadata (which can be signed).

SSO in {productname} does not support:

* Product choosing and implementation for the Identity Service Provider (IdP).

> * SAML support for other products (please, check with the respective product documentation).

Reviewer: ```suggestion
* SAML support for other products (check with the respective product documentation).
```

After the authentication with the IdP using the user `orgadmin` is successful, you will be logged in into {productname} as the `orgadmin` user, provided that the `orgadmin` user exists in {productname}.

> === Enabling SOO

Reviewer: ```suggestion
=== Enable SOO
```
